packages/utilities/src/styled.tsx

```diff
@@ -74,19 +72,24 @@ export function styled<TComponentProps extends IPropsWithStyles<TStyleProps, TSt
   return Wrapped as (props: TComponentProps) => JSX.Element;
 }
 
-function _resolve<TStyleProps, TStyles>(
+function _resolve<TStyleProps, TStyleSet extends IStyleSet<TStyleSet>>(
   styleProps: TStyleProps,
-  ...allStyles: (IStyleFunctionOrObject<TStyleProps, Partial<TStyles>> | undefined)[]
-): Partial<TStyles> | undefined {
-  const result: Partial<TStyles>[] = [];
+  ...allStyles: (IStyleFunctionOrObject<TStyleProps, TStyleSet> | undefined)[]
+): IConcatenatedStyleSet<TStyleSet> | undefined {
+  const result: Partial<TStyleSet>[] = [];
 
   for (const styles of allStyles) {
     if (styles) {
       result.push(typeof styles === 'function' ? styles(styleProps) : styles);
     }
   }
   if (result.length) {
-    return concatStyleSets(...result);
+    // cliffkoh: I cannot figure out how to avoid the cast to any here.
+    // It is something to do with the use of Omit in IStyleSet.
+    // It might not be necessary once  Omit becomes part of lib.d.ts (when we remove our own Omit and rely on
+    // the official version).
+    // tslint:disable-next-line:no-any
+    return concatStyleSets(...(result as any)) as IConcatenatedStyleSet<TStyleSet>;
```

Comment:
Could it be related to [this](https://github.com/Microsoft/TypeScript/issues/13557)? I feel like I had a [similar issue](https://github.com/OfficeDev/office-ui-fabric-react/blob/1dbfc5ee7c38e982282f13ef92884538e7226169/packages/foundation/src/createComponent.tsx#L64).